test/gpu/native/localeName.good

```diff
@@ -1,3 +1,4 @@
 warning: The prototype GPU support implies --no-checks. This may impact debuggability. To suppress this warning, compile with --no-checks explicitly
 
+LOCALE0.node0-GPU0
```

Comment:
Wait, does this means (at least for the GPU sublocale):

`writeln(here)`
`writeln(here.name)`

Return different values:
`LOCALE0.node0-GPU0` vs `node0-GPU0`

This seems really odd to me. Is this the behavior we get with non-GPU locales as well?

Reply:
Hmm... yes. We would get the same behavior with NUMA as `writeThis` writes the parent locale first than adds the `.name` at the end. That's what the GPU locale model is doing today. So, it is consistent. I don't know whether writeThis should write the name (as is the case for `flat` locale model, but that's more straightforward), or it should write something more complex. `LOCALE0.node0` sure feels redundant, though.

Maybe this is something to be discussed as part of the locale module stabilization? Pinging @jeremiah-corrado for that part.

Reply:
This relates somewhat to the thread about `locale.name` and `locale.hostname` from the module review. Specifically this comment: https://github.com/chapel-lang/chapel/issues/20223#issuecomment-1185774419, where I pointed out that `writeln(here)` and `writeln(here.name)` produce different output for a Flat locale. 

If the goal is to keep the behavior of these methods consistent (or at least analogous) between GPU locales and Flat locales, then I think the above behavior is okay because `locale.here` and `writeThis` *also* currently give different output for the Flat model. 

However, if this is not what we want for GPUs (i.e. we think that `here.gpus[0].name == stringify(here.gpus[0])` should be true), then I think we should also change the behavior of either `locale.name` or the `writeThis` implementation for Flat locales s.t. there is uniformity across models. 

I'm not sure which option is best thought. I'd be interested to see if anyone knows/remembers the rationale for making them different on the Flat locale model to begin with.

Reply:
> Are you OK with proceeding with this? Looks like we aren't doing anything different than other locale models. The thing we're doing might be questionable, but this one's a still progress, I'd say?

Oops, just saw this come in. Yeah for sure don't hold up because of this.